### PR TITLE
TGUIv4: TGchat fix traitor codeword highlighting

### PR DESCRIPTION
**Pull request by S34NW**
## What Does This PR Do
Fixes traitor codeword highlighting for TGchat, partially implements TG's system for this. But due to their different system for languages, speaking, and hearing, I had to sorta smoosh it in paracode in a bespoke way.

## Why It's Good For The Game
Feature parity for TGUIv3 and TGUIv4

## Changelog
fix: Fixed TGchat traitor codeword highlighting

### CODE REVIEW

File: code/__HELPERS/name_helpers.dm [hunk 1]
```diff
@@ -122,9 +126,12 @@ GLOBAL_VAR(syndicate_code_response) //Code response for traitors.
 	/N
 	*/
 
-/proc/generate_code_phrase()//Proc is used for phrase and response in master_controller.dm
+/proc/generate_code_phrase(return_list=FALSE)//Proc is used for phrase and response in master_controller.dm
```
Comment (DGamerL) [inline code/__HELPERS/name_helpers.dm:129]:
```suggestion
/proc/generate_code_phrase(return_list = FALSE) // Proc is used for phrase and response in master_controller.dm
```

File: code/__HELPERS/name_helpers.dm [hunk 2]
```diff
@@ -155,121 +162,35 @@ GLOBAL_VAR(syndicate_code_response) //Code response for traitors.
 				switch(rand(1,2))//Mainly to add more options later.
 					if(1)
 						if(names.len)
```
Comment (DGamerL) [inline code/__HELPERS/name_helpers.dm:164]:
```suggestion
						if(length(names))
```

File: code/__HELPERS/name_helpers.dm [hunk 3]
```diff
@@ -155,121 +162,35 @@ GLOBAL_VAR(syndicate_code_response) //Code response for traitors.
 				switch(rand(1,2))//Mainly to add more options later.
```
Comment (DGamerL) [inline code/__HELPERS/name_helpers.dm:162]:
```suggestion
				switch(rand(1, 2)) // Mainly to add more options later.
```

File: code/__HELPERS/name_helpers.dm [hunk 4]
```diff
@@ -155,121 +162,35 @@ GLOBAL_VAR(syndicate_code_response) //Code response for traitors.
...
-						code_phrase += pick(GLOB.verbs)
-		if(words==1)
-			code_phrase += "."
-		else
-			code_phrase += ", "
+						. += pick(GLOB.verbs)
 
-	return code_phrase
+		if(!return_list)
+			if(words==1)
```
Comment (DGamerL) [inline code/__HELPERS/name_helpers.dm:186]:
```suggestion
			if(words == 1)
```

File: code/__HELPERS/name_helpers.dm [hunk 5]
```diff
@@ -155,121 +162,35 @@ GLOBAL_VAR(syndicate_code_response) //Code response for traitors.
...
 				switch(rand(1,2))//Places or things.
 					if(1)
-						code_phrase += pick(drinks)
+						. += pick(drinks)
 					if(2)
-						code_phrase += pick(locations)
+						. += pick(locations)
 				safety -= 2
 			if(3)
 				switch(rand(1,3))//Nouns, adjectives, verbs. Can be selected more than once.
```
Comment (DGamerL) [inline code/__HELPERS/name_helpers.dm:177]:
```suggestion
				switch(rand(1, 3)) // Nouns, adjectives, verbs. Can be selected more than once.
```

**Review by AffectedArc07 — APPROVED** [review]
I dont know if I need to do it on this branch but here I am

**Pull request merged**